doc/vrrp/VRRP_Adaptation_HLD.md

```diff
@@ -0,0 +1,739 @@
+    - Config virtual MAC to Macvlan device;
+    - Add/del VIP to Macvlan device;
+    - Update VRRP instance configuration to the APPL DB;
+    - Update VRRP instance configuration to vrrpd by using vtysh commands.
+
+vrrpsyncd: 
+  - Complete the following tasks:
+    - Listen for kernel Macvlan device state change events, The state of the Macvlan device here determines the Master/Backup state of the VRRP instance, where up represents Master and down represents Backup.
+    - Update the kernel Macvlan device state to the APPL DB.
+
+##### BGP container
+
+zebra:
+  - Provide an API for vrrpd to update kernel Macvlan device state.
+  - Listen for kernel interface state change events and notify vrrpd.
+
+##### SWSS container
+
+intforch: 
+  - Subscribes to APPL_DB tables, responsible for updating the ASIC DB. Program the VIP and Virtual RIF object via SAI API.
+
+#### CoPP Configurations
+
+CoPP will be extended as follows for trapping VRRPs. Whether to install it depends on the FEATURE|vrrp table in Config DB. If the VRRP state is enabled, it will be installed; otherwise, it will not be installed.
+```
+  "trap.group.vrrp": {
+    "cir":"300",
+    "cbs":"300",
+    "meter_type":"packets",
+    "mode":"sr_tcm",
+    "red_action":"drop",
+    "trap_action":"trap",
+    "trap_priority":"7",
+    "queue": "7",
+    "yellow_action":"drop"
+  },
+  ...
+  "vrrp": {
+    "trap_ids": "vrrp", "vrrpv6",
+    "trap_group": "trap.group.vrrp"
+  },
+```
+
+#### DB Changes
+
+##### CONFIG_DB changes
+
+VRRP
+
+Producer: config manager
+
+Consumer: vrrpmgrd
+
+Description: New table that stores VRRP configuration for per interface + VRID.
+```
+;New table
+;holds the VRRP configuration per interface and VRID
+key             = VRRP:interface_name:vrid
+                                                ; Interface name string. Vlan, Ethernet, sub-interfaces or PortChannel
+                                                ; vrid is an integer
+; field = value
+vip                   = ip_address                    ; Virtual IP address. This is a list of IP addresses
+priority              = 3DIGIT                        ; Priority of VRRP instance
+adv_interval          = 5DIGIT                        ; Advertisement interval for VRRP. Default = 1000ms
+admin_status          = "up" | "down"                 ; String denoting the state of VRRP instance admin status. Default is up.
+version               = vrrp_version                  ; VRRP version. Value is 2 or 3. Default is 3
+preempt               = "enabled"/"disabled"          ; VRRP pre-emption is enabled? Default is enabled
+track_interface       = interface_name                ; This is a list of configured tracking interfaces
+priority_decrement    = 2DIGIT                        ; Specifies how much to decrement the priority of the VRRP instance if the tracking interface goes down, Default is 20
+```
```

Comment:
Please take the CONFIG_DB, APP_DB section from VRRP_HLD (keepalived based) sec 4.2.1

- Elaborate on track interface format
-

Reply:
done